core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/AddOperationsToChain.java

```diff
@@ -0,0 +1,234 @@
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.store.operationdeclaration.OperationDeclarations;
+import uk.gov.gchq.gaffer.user.User;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A <code>AddOperationsToChain</code> is a {@link GraphHook} that allows a
+ * user to insert additional operations at certain points on the operation chain.
+ * At the start, before a specific Operation, after a specific Operation, or at the end.
+ * A user can also specify authorised Operations to add, and if the user has
+ * the opAuths, the additional Operations will be added to the chain.
+ */
+public class AddOperationsToChain implements GraphHook {
+    private static final JSONSerialiser JSON_SERIALISER = new JSONSerialiser();
+
+    private AdditionalOperations defaultOperations = new AdditionalOperations();
+    private LinkedHashMap<String, AdditionalOperations> authorisedOps;
+
+    /**
+     * Default constructor.
+     */
+    public AddOperationsToChain() {
+    }
+
+    public void setStart(final List<Operation> start) {
+        this.defaultOperations.setStart(start);
+    }
+
+    public List<Operation> getStart() {
+        return defaultOperations.getStart();
+    }
+
+    public void setEnd(final List<Operation> end) {
+        this.defaultOperations.setEnd(end);
+    }
+
+    public List<Operation> getEnd() {
+        return defaultOperations.getEnd();
+    }
+
+    public void setBefore(final Map<String, List<Operation>> before) {
+        this.defaultOperations.setBefore(before);
+    }
+
+    public Map<String, List<Operation>> getBefore() {
+        return defaultOperations.getBefore();
+    }
+
+    public void setAfter(final Map<String, List<Operation>> after) {
+        this.defaultOperations.setAfter(after);
+    }
+
+    public Map<String, List<Operation>> getAfter() {
+        return defaultOperations.getAfter();
+    }
+
+    public void setDefaultOperations(final AdditionalOperations defaultOperations) {
+        this.defaultOperations = defaultOperations;
+    }
+
+    public AdditionalOperations getDefaultOperations() {
+        return defaultOperations;
+    }
+
+    public void setAuthorisedOps(final LinkedHashMap<String, AdditionalOperations> authorisedOps) {
+        this.authorisedOps = authorisedOps;
+    }
+
+    public LinkedHashMap<String, AdditionalOperations> getAuthorisedOps() {
+        return authorisedOps;
+    }
+
+    /**
+     * Constructs a {@link AddOperationsToChain} using the path to json
+     * file specifying Operations to add.
+     *
+     * @param addOperationsPath Path to file containing Operations to add.
+     * @throws IOException if the file reading fails.
+     */
+    public AddOperationsToChain(final String addOperationsPath) throws IOException {
```

Comment:
Please can you put all the constructors together at the top of the class (below the fields).